### DISCUSSION TITLE
Modelling thrust vectoring and destructible components for a spaceship

### DISCUSSION THREAD

**Discussion by adihodos** [DanielChappuis/reactphysics3d#299] [Q&A]
To continue the discussion I started on the issues section. The main thing I want to do is to somehow to model all the exhaust nozzles separately and apply forces at the nozzles, not at the center of mass. So  I need to model somehow the rigid body as having a center fuselage, 4 wings, 4 engines and 16 nozzles that can apply force on the engines. I don't know how to do that. This will also make it possible to have parts of the ship be destructible and blown off like say for example get hit in a wing and lose half of it along with the engine.

**Comment by DanielChappuis** [comment]
Hello.

Once you a have created a RigidBody for your spaceship, you should be able to apply forces and torques onto it.
Take a look at all the methods available on the RigidBody class [here in the documentation](https://reactphysics3d.com/documentation/api/html/classreactphysics3d_1_1_rigid_body.html).

For instance, you can apply a local force at the center of mass location with:

rp3d::Vector3 force(4, 0, 0);
spaceShipBody->applyLocalForceAtCenterOfMass(force);
There are other methods to apply a force/torque at different location of your body.

Note: Can I please ask you to post your questions in the [Discussions](https://github.com/DanielChappuis/reactphysics3d/discussions) section instead of creating a new issue?
